src/proxy/outbound.rs

```diff
@@ -789,4 +786,21 @@ mod tests {
         gateway: &'a str,
         request_type: RequestType,
     }
+
+    pub fn test_store(xds_workloads: Vec<XdsWorkload>) -> anyhow::Result<WorkloadInformation> {
```

Comment:
same as above. not sure it needs to be pub any more and I think the name could be less ambiguous about what it's doing.
```suggestion
    fn new_mock_store(xds_workloads: Vec<XdsWorkload>) -> anyhow::Result<WorkloadInformation> {
```